portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,42 +37,57 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
-
   Gson gson = new Gson();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-    Query query = new Query("Task");
-
+    int userInput = getUserInput(request);
+    if (userInput == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter a non-negative integer.");
```

Comment:
Instead of this, your API should throw an error.

For example: `response.setStatus(HttpServletResponse.SC_BAD_REQUEST);`